qa/tasks/cephfs/test_snapshots.py

```diff
@@ -442,3 +444,51 @@ def test_multimds_hardlink(self):
 
         self.mount_a.run_shell(["rmdir", "d1/.snap/s1"])
         self.mount_a.run_shell(["rm", "-rf", "d0", "d1"])
+
+    def delete_dir_and_snaps(self, dir_name, snaps):
+        for sno in range(1, snaps+1, 1):
+            sname = "{dir_name}/.snap/s_{sno}".format(dir_name=dir_name, sno=sno)
+            self.mount_a.run_shell(["rmdir", sname])
+        self.mount_a.run_shell(["rmdir", dir_name])
+
+    def create_dir_and_snaps(self, dir_name, snaps):
+        self.mount_a.run_shell(["mkdir", dir_name])
+        
+        # creating first 100 snapshots must be successful
+        for sno in range(1, snaps+1, 1):
+            sname = "{dir_name}/.snap/s_{sno}".format(dir_name=dir_name, sno=sno)
+            try:
+                self.mount_a.run_shell(["mkdir", sname])
+            except CommandFailedError as e:
+                # failing at the last mkdir is expected
+                if sno == snaps:
+                    log.info("failed while creating snap #{}: {}".format(sno, repr(e)))
+                    sys.exc_clear()
+
+
+    def test_mds_max_snaps_per_dir(self):
+        """
+        Test the newly introudced option named mds_max_snaps_per_dir
+        Default snaps limit is 100
+        """
+        self.fs.set_allow_new_snaps(True)
+        status = self.fs.wait_for_daemons()
+        self.create_dir_and_snaps("accounts", 101)
+        self.delete_dir_and_snaps("accounts", 100)
+
```

Comment:
Then I recommend adding more tests:

- That increasing the snapshot limit allows a previously blocked snapshot (without restarting the MDS)
- That reducing the snapshot limit below the current number of snapshots works and prevents new snapshots from being created.

Reply:
Just noting these tests need written still.